storage/statedb/proof.go

```diff
@@ -135,7 +136,444 @@ func VerifyProof(rootHash common.Hash, key []byte, proofDB database.DBManager) (
 	}
 }
 
-func get(tn node, key []byte) ([]byte, node) {
+// proofToPath converts a merkle proof to trie node path. The main purpose of
+// this function is recovering a node path from the merkle proof stream. All
+// necessary nodes will be resolved and leave the remaining as hashnode.
+//
+// The given edge proof is allowed to be an existent or non-existent proof.
+func proofToPath(rootHash common.Hash, root node, key []byte, proofDb database.DBManager, allowNonExistent bool) (node, []byte, error) {
```

Comment:
Hmm.. Just curious that do we need to rebuild a state trie?
Is this function called during the block execution? If yes, then we can reuse current state trie.

Reply:
@kjhman21 No, this function is required when generating a snapshot initially, not the block execution.

Reply:
got it. Thanks.